### ISSUE TITLE
bundler is scanning for all possible gemspec

### ISSUE THREAD

**Issue by mathieujobin** [bundler/bundler#6216]
finding `bundle update` has gotten very slow... 

found this https://dependabot.com/blog/improving-dependency-resolution-in-bundler

seems like a PR into Molinillo was merged, I'm not sure it made its way to bundler 1.16.0 and I also don't understand how cocaopods and ruby a mixed together...
but before downgrading to bundler 1.14

I ran bundle update with strace, and found out it was looking for A LOT of outdated gemspec

I am thinking this could be the cause of the slowness

try out this command in your project

`strace bundle update 2>&1 | grep "^stat.*gemspec.*ENOENT" > omg.log`

and `wc -l omg.log`

here is a very short snippet of the output

```
stat("/home/mathieu/.gem/specs/rspec-3.5.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.4.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.4.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.3.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.3.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.2.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.2.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.1.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.1.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.0.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.0.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.99.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.99.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.14.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.14.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.14.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.14.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.13.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.13.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.12.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.12.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.11.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.11.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.10.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.10.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.9.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.9.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.8.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.8.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.7.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.7.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.6.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.6.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.5.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.5.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.4.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.4.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.3.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.3.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.2.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.2.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.1.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.1.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.6.0.beta2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.6.0.beta2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.6.0.beta1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.6.0.beta1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.5.0.beta4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.5.0.beta4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.5.0.beta3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.5.0.beta3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.5.0.beta2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.5.0.beta2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.5.0.beta1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.5.0.beta1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.0.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.0.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.0.0.beta2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.0.0.beta2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-3.0.0.beta1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-3.0.0.beta1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.99.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.99.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.99.0.beta2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.99.0.beta2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.99.0.beta1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.99.0.beta1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.14.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.14.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.9.0.rc2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.9.0.rc2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.8.0.rc2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.8.0.rc2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.8.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.8.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.7.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.7.0.rc1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.6.0.rc6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.6.0.rc6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.6.0.rc4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.6.0.rc4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.6.0.rc2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.6.0.rc2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.3.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.3.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.3.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.3.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.3.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.3.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.2.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.2.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.12.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.12.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.11.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.11.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.1.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.1.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.0.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.0.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.9.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.9.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.9.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.9.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.9.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.9.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.9.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.9.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.9.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.9.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.8.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.8.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.8.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.8.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.8.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.8.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.7.5.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.7.5.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.7.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.7.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.7.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.7.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.7.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.7.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.7.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.7.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.7.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.7.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.7.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.7.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.6.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.6.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.6.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.6.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.6.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.6.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.6.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.6.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.6.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.6.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.16.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.16.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.15.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.15.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.14.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.14.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.13.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.13.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.12.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.12.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.11.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.11.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.5.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.5.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.4.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.4.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.3.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.3.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.3.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.3.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.3.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.3.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.2.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.2.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.1.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.1.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.1.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.1.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.1.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.1.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.1.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.1.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.1.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.1.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.1.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.1.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.1.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.1.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.1.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.1.0.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-0.0.10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-0.0.10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.rc.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.rc.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.22.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.22.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.20.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.20.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.19.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.19.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.18.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.18.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.17.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.17.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.16.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.16.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.15.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.15.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.14.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.14.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.13.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.13.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.12.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.12.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.11.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.11.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.beta.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.beta.1.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a10.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a9.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a8.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a7.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a6.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a5.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a4.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a3.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-2.0.0.a2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-2.0.0.a2.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.2/specifications/rspec-1.3.1.rc.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/rspec-1.3.1.rc.gemspec", 0x7ffe1f4b1180) = -1 ENOENT (No such file or directory)
```

you can see it is trying to find all possible old version of the rspec gem ...

**Comment by mathieujobin** [comment]
```
$ wc -l omg.log 
10902 omg.log
```

```
$ grep -o "[^/]*gemspec" omg.log | cut -d - -f 1 | sort | uniq -c  | sort -n | tail -n 30
    124 vcr
    126 json
    132 coderay
    132 faraday
    136 uglifier
    140 tzinfo
    142 activejob
    142 byebug
    142 capybara
    146 rake
    164 therubyracer
    166 nokogiri
    170 rack
    182 excon
    182 redis
    204 webmock
    220 actionview
    234 unicorn
    260 cucumber
    272 rollbar
    284 sidekiq
    290 haml
    350 oj
    426 newrelic_rpm
    428 pry
    502 railties
    612 activesupport
    640 actionpack
    668 activerecord
    686 rspec
```

There must be a condition that could be used to end the scan loop earlier?

**Comment by mathieujobin** [comment]
or maybe it can be cached?

**Comment by segiddins** [comment]
Can you share a gemfile we can use to reproduce? Thanks

**Comment by mathieujobin** [comment]
sure, stay tuned

**Comment by mathieujobin** [comment]
since my gem file has several private gems, I could not share in its entirety.
I have taken out several gems and put them back one by one until I locate the culprit.
and i found out it happens when i load `sidekiq-pro` from a `source` block

so when I don't do a block such as 

```
source "https://LOGIN:PASWORD@gems.contribsys.com/"

gem 'sidekiq-pro', '~> 3.0'
```

it won't try to find all old possible versions of every gems

but if I do

```
source "https://LOGIN:PASWORD@gems.contribsys.com/" do
  gem 'sidekiq-pro', '~> 3.0'
end
```

it leads to 2800 unsuccessful file look up

**Comment by mathieujobin** [comment]
[Gemfile.txt](https://github.com/bundler/bundler/files/1594146/Gemfile.txt)

**Comment by segiddins** [comment]
I get:

```
Authentication is required for gems.contribsys.com.
Please supply credentials for this source. You can do this by running:
 bundle config gems.contribsys.com username:password
```

**Comment by mathieujobin** [comment]
yeah, I'm sorry I can't share this login/password

let me figure out a different way to reproduce...

**Comment by mperham** [comment]
@segiddins Happy to give you a temporary credential so you can optimize this case, email mperham at gmail.  My server uses the old school Rubygems static file protocol and not the Bundler API.

**Comment by mperham** [comment]
@mathieujobin As a workaround, it will  likely be much faster if you locked to a specific version.  Using `~> 3.0` is nebulous and likely triggers a version search.

**Comment by heironimus** [comment]
@mperham any chance you plan to update to the Bundler API?

**Comment by mperham** [comment]
@heironimus No plans at the moment.

**Comment by segiddins** [comment]
I also can't run `strace` since I'm on macOS

**Comment by aledalgrande** [comment]
@segiddins you can use `dtrace` though? Needs `sudo` I think.

**Comment by segiddins** [comment]
@aledalgrande I probably could, but I don't know how, and when I tried last time I spent a few hours and got nowhere, either with dtrace or reproducing the issue, and I don't remember which

**Comment by mathieujobin** [comment]
we switch from `rubygems.org` to `.jfrog.io` and its a lot better

removing all of our `ssh` sourced gems.

now that same command from above, only complains about sidekiq-pro, which is much less

```
[mathieu@hz550 push_it]$ cat today-omg.log 
stat("api_hammer.gemspec", 0x7ffe84d9a1d0) = -1 ENOENT (No such file or directory)
stat("api_hammer.gemspec", 0x7ffe84d9a3a0) = -1 ENOENT (No such file or directory)
stat("api_hammer.gemspec", 0x7ffe84d9a3b0) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.3/specifications/sidekiq-pro-3.7.1.gemspec", 0x7ffe84d99980) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/sidekiq-pro-3.7.1.gemspec", 0x7ffe84d999f0) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.3/specifications/sidekiq-pro-3.6.1.gemspec", 0x7ffe84d99a60) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/sidekiq-pro-3.6.1.gemspec", 0x7ffe84d99a60) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.3/specifications/sidekiq-pro-3.6.0.gemspec", 0x7ffe84d99a60) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/sidekiq-pro-3.6.0.gemspec", 0x7ffe84d99a60) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.rvm/gems/ruby-2.4.3/specifications/sidekiq-pro-3.7.0.pre1.gemspec", 0x7ffe84d99a60) = -1 ENOENT (No such file or directory)
stat("/home/mathieu/.gem/specs/sidekiq-pro-3.7.0.pre1.gemspec", 0x7ffe84d99a60) = -1 ENOENT (No such file or directory)
[mathieu@hz550 push_it]$ 
```

```
$ wc -l omg.log today-omg.log 
  10902 omg.log
     11 today-omg.log
```

I'm happy with that

**Comment by mathieujobin** [comment]
@mperham oh, and using `~> 3.7.0` or `~> 3.0` did not change anything

**Comment by mperham** [comment]
The issue is that the old school rubygems provider in Bundler is downloading each version's gemspec.rz in serial.  I have something like 50 Sidekiq Pro versions from 1.0 onward, it downloads 1.0.0, 1.1.0, 1.1.1, ... until the current 4.0.2.  Each request takes 100+ ms round trip,  and it does it for sidekiq-ent too and it does it again, after printing this:

```
Double checking for all specs (due to the size of the request) in rubygems repository https://[snip].contribsys.com/, https://rubygems.org/ or installed locally
```

All of this adds up to a 30 second `bundle up sidekiq-ent`.  Removing my server, `bundle up` drops to 3 seconds.  Two things to possibly fix:

1. Remove the "double check" if possible, should halve the overhead.
2. Download the gemspec.rz files in parallel, with a reasonable (say, 10) concurrency

Happy to open a new issue if this is something the Bundler team is interested in tackling.

**Comment by greysteil** [comment]
I don't think this is grouping logic - sounds like @mperham has it right that it's an issue with `gems.contribsys.com` not serving the compact index (and instead Bundler having to fall back to the old API).

The introduction of the slowness with Bundler 1.16.0 makes sense because before resolution would have "given up" early on some graphs, whereas now it's more dogged. That could easily have introduced cases like this one where previously it was only ever trying the latest `sidekiq-ent`, and not having to making lots of other requests.

I think the solution here is something along what @mperham suggests. Have we looked at adding parallelism to gemspec downloads before?

**Comment by segiddins** [comment]
I don’t think so? We do have parallelism for compact index downloads though... that might not be a bad idea? I guess we could also cache the downloaded gemspecs, though that sounds like a larger project

**Comment by mperham** [comment]
@segiddins I noticed that each gemspec is downloaded **twice** from my server for a `bundle up`.  It doubles the execution time.  Caching across `bundle` executions is troublesome but seems useful here within the same operation.

**Comment by segiddins** [comment]
> I noticed that each gemspec is downloaded twice from my server for a bundle up. It doubles the execution time

That's _definitely_ a bug. If you could open up a new issue with a Gemfile I can use to reproduce, I'd be happy to look into fixing it
